### PR TITLE
issue 28230 short lived permission cache2

### LINKED ISSUE #28230: spike(performance): test short lived permission cache
### Problem

With large content pulls and limited users, this method can consume a lot of CPU time.  The reason for this is that EVERY content pull in dotCMS is filtered through permissions, which are both hierarchical and recursive.  For example, when rendering a front end page in dotCMS

Here is the ticket for reference:
https://dotcms.zendesk.com/agent/tickets/115552

The idea is that a short lived cache on the lookup `doesUserHavePermission` can improve performance.

### PR DESCRIPTION
**Pull request by wezell**
- **spike(performance) try a short lived cache on top of doesUserHavePermission**
- **spike(performance) adding null checks because... we do nulls :( ref:#28230**
- **spike(performance) adding null checks because... we do nulls :( ref:#28230**
- **#28230 caching permissions**
- **#28230 short lived permission cache for read permissions**

### CODE REVIEW

File: dotcms-integration/docker-compose.yml
Comment (wezell) [inline dotcms-integration/docker-compose.yml:1]: This file is not used by anything.

File: dotcms-integration/pom.xml
Comment (wezell) [inline dotcms-integration/pom.xml:1]: These values are reveresed

File: docker-compose.yml
Comment (wezell) [inline docker-compose.yml:1]: this file is not used by anything

**Review by erickgonzalez — APPROVED** [review]

**Review by swicken-dotcms — APPROVED** [review]
Looks good to me

**Pull request merged**